discovery/dns/dns.go

```diff
@@ -317,19 +317,26 @@ func askServerForName(name string, queryType uint16, client *dns.Client, servAdd
 	}
 
 	response, _, err := client.Exchange(msg, servAddr)
-	if err == dns.ErrTruncated {
+	if err != nil {
+		if err == dns.ErrId {
+			if response == nil { // response can be nil here, just be paranoid
```

Comment:
@erikh , I still think this should be removed. Look at L431 of `dns/client.go` [here](https://github.com/miekg/dns/blob/a288d199dc686dca98438c25c90c16647a6e28df/client.go#L431). For the outcome of ErrId, a nil of check of `response` is already being done without a guard. If `response` was nil , then the panic would originate from the `miekg/dns` itself. Now if it so happens that an unguarded nil check is being in that library, then that fix should go upstream :).

Reply:
So based on my comment above: if `response` was indeed nil, the `panic` would happen anyway and originate from the library. This guard will not protect us unfortunately.